components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/EmailTemplateManagerImpl.java

```diff
@@ -477,33 +510,107 @@ private String resolveNotificationChannel(String notificationChannel) {
         }
     }
 
+    /**
+     * Create a registry resource instance of the notification template.
+     *
+     * @param notificationTemplate Notification template
+     * @return Resource
+     * @throws NotificationTemplateManagerServerException Error while creating the resource
```

Comment:
@throws NotificationTemplateManagerServerException If an error occurred while creating the resource.